src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -379,14 +650,12 @@ class SegmentCleaner : public SegmentProvider {
   const bool detailed;
   const config_t config;
 
-  segment_id_t num_segments = 0;
-  size_t segment_size = 0;
-  size_t block_size = 0;
+  internal_segment_id_t num_segments = 0;
 
   ExtentReaderRef scanner;
 
   SpaceTrackerIRef space_tracker;
-  std::vector<segment_info_t> segments;
+  segment_info_set_t<segment_info_t> segments;
```

Comment:
can we rename segments to seg_infos  or segment_infos, always get confused when code has segment variable which stand for segment_id_t.